src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -75,12 +78,14 @@ def __init__(
         file_path: str | None = None,
         file_path_description_of_origin: str | None = None,
         is_lockfile: bool = False,
+        lockfile_hex_digest: str | None = None,
```

Comment:
You can remove `is_lockfile`. If `lockfile_hex_digest` is set, that implies it's a lockfile